docs/guide/repository.md

```diff
@@ -1549,6 +1549,124 @@ class AcmeFooBundle extends Bundle
+
+```php
+<?php
+
+/**
+ * Interface for LimitationType form mappers.
+ *
+ * It maps a LimitationType's supported values to editing form.
+ */
+interface LimitationFormMapperInterface
+{
+    /**
+     * "Maps" Limitation form to current LimitationType, in order to display one or several fields
+     * representing limitation values supported by the LimitationType.
+     *
+     * Implementors MUST either:
+     * - Add a "limitationValues" form field
+     * - OR add field(s) that map to "limitationValues" property from $data.
+     *
+     * @param FormInterface $form Form for current Limitation.
+     * @param Limitation $data Underlying data for current Limitation form.
+     */
+    public function mapLimitationForm(FormInterface $form, Limitation $data);
+    /**
+     * Returns the Twig template to use to render the limitation form.
+     *
+     * @return string
+     */
+    public function getFormTemplate();
+    /**
+     * This method will be called when FormEvents::SUBMIT is called.
+     * It gives the opportunity to filter/manipulate limitation values.
+     *
+     * @param Limitation $limitation
+     */
+    public function filterLimitationValues(Limitation $limitation);
+}
+```
+
+Next, register the service in DIC with the `ez.limitation.formMapper` tag and set the `limitationType` attribute to the Limitation type's identifier:
+
+```yml
+acme.security.limitation.custom_limitation.mapper:
+    class: 'Acme\FooBundle\Security\Limitation\Mapper\CustomLimitationFormMapper'
+    arguments:
+        # ...
+    tags:
+        - { name: 'ez.limitation.formMapper', limitationType: 'Custom' }
+```
+
+If you want to provide human-readable names of the custom Limitation values, you need to implement [`\EzSystems\RepositoryForms\Limitation\LimitationValueMapperInterface`](https://github.com/ezsystems/repository-forms/blob/master/lib/Limitation/LimitationValueMapperInterface.php):
+
+```php
+<?php
+
+/**
+ * Interface for Limitation Value mappers.
+ */
+interface LimitationValueMapperInterface
+{
+    /**
+     * Map the limitation values, in order to pass them as context of limitation value rendering.
+     *
+     * @param Limitation $limitation
+     * @return mixed[]
+     */
+    public function mapLimitationValue(Limitation $limitation);
+}
+```
+
+Then register the service in DIC with the `ez.limitation.valueMapper` tag and set the `limitationType` attribute to Limitation type's identifier:
+
+```yml
+acme.security.limitation.custom_limitation.mapper:
+    class: 'Acme\FooBundle\Security\Limitation\Mapper\CustomLimitationValueMapper'
+    arguments:
+        # ...
+    tags:
+        - { name: 'ez.limitation.valueMapper', limitationType: 'Custom' }
+```
+
+If you want to completely override the way of rendering of custom Limitation values in the role view, you need to create a Twig template containing block definition which follows the naming convention: `ez_limitation_<LIMITATION TYPE>_value`. For example:
```

Comment:
the way of rendering **of** custom Limitation values -> the way of rendering custom Limitation values